Mage.Sets/src/mage/cards/b/BewitchingLeechcraft.java

```diff
@@ -0,0 +1,109 @@
+        this.subtype.add(SubType.AURA);
+
+        // Enchant creature
+        TargetPermanent auraTarget = new TargetCreaturePermanent();
+        this.getSpellAbility().addTarget(auraTarget);
+        this.getSpellAbility().addEffect(new AttachEffect(Outcome.BoostCreature));
+        this.addAbility(new EnchantAbility(auraTarget));
+
+        // When Bewitching Leechcraft enters the battlefield, tap enchanted creature.
+        this.addAbility(new EntersBattlefieldTriggeredAbility(new TapEnchantedEffect()));
+
+        // Enchanted creature has "If this creature would untap during your untap step,
+        // remove a +1/+1 counter from it instead. If you do, untap it."
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD,
+            new GainAbilityAttachedEffect(
+                new SimpleStaticAbility(new BewitchingLeechcraftReplacementEffect()),
+                AttachmentType.AURA
+            )
+        ));
+    }
+
+    private BewitchingLeechcraft(final BewitchingLeechcraft card) {
+        super(card);
+    }
+
+    @Override
+    public BewitchingLeechcraft copy() {
+        return new BewitchingLeechcraft(this);
+    }
+}
+
+class BewitchingLeechcraftReplacementEffect extends ReplacementEffectImpl {
+
+    public BewitchingLeechcraftReplacementEffect() {
+        super(Duration.EndOfGame, Outcome.Detriment);
+        staticText = "If this creature would untap during your untap step, " +
+            "remove a +1/+1 counter from it instead. If you do, untap it.";
+    }
+
+    public BewitchingLeechcraftReplacementEffect(final BewitchingLeechcraftReplacementEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public BewitchingLeechcraftReplacementEffect copy() {
+        return new BewitchingLeechcraftReplacementEffect(this);
+    }
+
+    @Override
+    public boolean checksEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.UNTAP
+            && game.getTurnStepType() == PhaseStep.UNTAP;
+    }
+
+    @Override
+    public boolean replaceEvent(GameEvent event, Ability source, Game game) {
+        Permanent permanentUntapping = game.getPermanent(source.getSourceId());
+        if(!applies(event,source,game)){
+            return false;
+        }
+
+        // If we could not remove a counter, we are replacing the UNTAP event.
+        // If we could remove a counter, we are not replacing the UNTAP, just adding to it.
+        return !permanentUntapping.getCounters(game).removeCounter(CounterType.P1P1, 1);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        return false;
+    }
+
+    @Override
+    public boolean applies(GameEvent event, Ability source, Game game) {
+        Permanent eventPermanent = game.getPermanent(event.getTargetId());
+        Permanent sourcePermanent = game.getPermanent(source.getSourceId());
+        return eventPermanent != null && sourcePermanent != null && sourcePermanent == eventPermanent;
```

Comment:
Not good practice to compare `Permanent` with `==` operator. Need to compare their UUIDs with `.equals()`. I think you can just use `event.getTargetId().equals(source.getSourceId()))`